lib/appRelationsDiscovery.js

```diff
@@ -69,6 +70,8 @@ async function getRelations(asyncApiDocs, { syntax } = defaultOptions) {
   });
   if (syntax === 'default')
     return metrics;
+  if (syntax === 'mermaid')
+    return getMermaidDiagram(metrics);
```

Comment:
You can use here the `switch` statement for better maintanance :)

... but I think that better will be split the functions for changing default output to another by separate functions. What do I mean?

At the moment user can use `getRelations` function and pass options which format he/she needs, but we can change this behaviour and always return default syntax and then export from package the `getMermaidDiagram` function, so usage of package will look like:

```js
const output = getRelations(...asyncAPIDocs);
const mermaid = getMermaidDiagram(output);
```

We can make in as the followup, but better will be making the issue to discuss it (we can wait for Łukasz). What do you think of this approach? Of course we can stay with current logic at the moment :)

Reply:
> You can use here the `switch` statement for better maintanance :)

I had it in my mind but thought that I would change it after we start adding more custom syntax :)

> I think that better will be split the functions for changing default output to another by separate functions

Yeah, we could have that. When we will support class models for our metrics we could have APIs which could add a subscriber or publisher to the channel to see what happens with the system if one of the puzzles is removed/added.

```javascript
const output = getRelations(...asyncAPIDocs);
output.subscribe(<channelName>,<server>,<serviceName>);
const mermaid = getMermaidDiagram(output);
```

> but better will be making the issue to discuss it (we can wait for Łukasz). What do you think of this approach? Of course we can stay with current logic at the moment :)

Sure, will open it up :)

Reply:
Ok, so we can stay with the current solution and wait for Łukasz :)